.github/workflows/docker-build-deploy.yml

```diff
@@ -0,0 +1,65 @@
+name: build & deploy Docker image
+
+on:
+  push:
+    branches:
+      - deploy
+      # todo: remove
+      - gh-action
+  # pull_request:
+  #   branches:
+  #     - deploy
+
+env:
+  DOCKER_IMAGE: registry.gitlab.tpwd.de/tpwd/bb-navi/amarillo
+  DOCKER_TAG: production
+
+jobs:
+  prep:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - uses: docker/setup-buildx-action@v1
+    - name: log in to registry.gitlab.tpwd.de
+      uses: docker/login-action@v1
+      with:
+        registry: registry.gitlab.tpwd.de
+        username: ${{ secrets.GITLAB_USERNAME }}
+        password: ${{ secrets.GITLAB_PASSWORD }}
+    - name: build & push Docker image
+      uses: docker/build-push-action@v2
+      with:
+        push: true
+        tags: ${{ env.DOCKER_IMAGE }}:${{ env.DOCKER_TAG }}
+        # https://github.com/docker/build-push-action/blob/9472e9021074a3cb3279ba431598b8836d40433f/docs/advanced/cache.md#github-cache
+        # https://github.com/moby/buildkit#registry-push-image-and-cache-separately
+        cache-from: type=gha
+        cache-to: type=gha,mode=max,oci-mediatypes=true,compression=zstd
+  deploy:
+    runs-on: ubuntu-latest
```

Comment:
As RIDE2GO_TOKEN is not an organisation secret, but in stored in the repo environment `production`, this needs to be added explicitly, see [here](https://docs.github.com/en/actions/deployment/targeting-different-environments/using-environments-for-deployment#using-an-environment)

```suggestion
    runs-on: ubuntu-latest
    environment: production
```